src/sentry/api/endpoints/organization_transaction_details.py

```diff
@@ -0,0 +1,363 @@
+    query.columns += [
+        Column("tags.key"),
+        Column("tags.value"),
+        Column("sentry_tags.key"),
+        Column("sentry_tags.value"),
+    ]
+    results = query.run_query(referrer=Referrer.API_ORGANIZATION_TRANSACTION_DETAILS.value)
+    return results["data"]
+
+
+def _span_data_to_event_spans(span_data):
+    def _normalize_group(group):
+        if group == "00":
+            return None
+        return group
+
+    return [
+        {
+            "timestamp": span["precise.finish_ts"],
+            "start_timestamp": span["precise.start_ts"],
+            "exclusive_time": span["exclusive_time"],
+            "description": span["description"],
+            "op": span["op"],
+            "span_id": span["span_id"],
+            "parent_span_id": span["parent_span_id"],
+            "trace_id": span["trace_id"],
+            "tags": dict(zip(span["tags.key"], span["tags.value"])),
+            "sentry_tags": dict(zip(span["sentry_tags.key"], span["sentry_tags.value"])),
+            "hash": _normalize_group(span["group"]),
+            "same_process_as_parent": span.get("same_process_as_parent"),
+        }
+        for span in span_data
+    ]
+
+
+def _span_ops_breakdown(spans_data):
+    # Roughly replicates the logic in Relay. See:
+    # https://github.com/getsentry/relay/blob/b2fcde7ddb829e53f8b312bc25b2dc24eaae3b84/relay-event-normalization/src/normalize/breakdowns.rs#L87
+    known_ops = {"db", "http", "resource", "browser", "ui"}
+    intervals_by_op = defaultdict(list)
+
+    for span in spans_data:
+        # Normalize known span ops, since those get reported individualy
+        span_op = span.get("op", "")
+        for op in known_ops:
+            if span_op.startswith(op):
+                span_op = op
+                break
+
+        intervals_by_op[span_op].append(
+            (span.get("precise.start_ts"), span.get("precise.finish_ts"))
+        )
+
+    total_time = 0.0
+    breakdown = {}
+    for op, intervals in intervals_by_op.items():
+        duration = _duration_from_intervals(intervals)
+        total_time += duration
+        if op in known_ops:
+            breakdown[f"ops.{op}"] = {
+                "value": round(duration, 3),
+                "unit": "milliseconds",
+            }
+
+    breakdown["total.time"] = {
+        "value": round(total_time, 3),
+        "unit": "milliseconds",
+    }
+    return breakdown
+
+
+def _duration_from_intervals(intervals):
+    intervals.sort(key=lambda x: x[0])
+
+    duration = 0.0
+    i = 0
+    while i < len(intervals):
+        start, end = intervals[i]
+        for j in range(i, len(intervals)):
+            next_span_start, next_span_end = intervals[j]
+            if next_span_start < end:
+                end = next_span_end
+                i = j
+        duration += (end - start) * 1000.0  # convert to ms
+        i += 1
+
+    return duration
+
+
+def _add_comparison_to_event(data, organization_id, project, average_columns):
```

Comment:
This is mostly identical to the `add_comparison_to_event` function in `src/sentry/api/endpoints/organization_event_details.py` right? Can we reuse the code between these two? At the very least the actual QueryBuilder/Running part?